### PR TITLE
[PLATFORM-449] Update Marketplace contract

### PR DESCRIPTION
**Pull request by fonty1**
**
Update:
- MP2 contract added to staging: https://github.com/streamr-dev/streamr-infra/wiki/Private-Ethereum-Chain
- Contract ABIs should be compatible, but in practice there seems to be issues publishing, so the old ABI is still used with the old contract and the new ABI is used with the new contract. 
- `requiresWhitelist` input removed from `createProduct` in the contract. There's now a separate endpoint for creating with whitelist.

**

---

New ABI, new MP address. I also added an ENV variable to toggle usage of the new contract: `NEW_MP_CONTRACT`. Try it out with: `NEW_MP_CONTRACT=on`. 

Worth mentioning that the new ABI is compatible with the old contract (but not the other way around). I.e. Function overloading OK, but underloading(?) causes problems. 

The contract Diff is explained in: https://streamr.atlassian.net/browse/PLATFORM-449
TLDR:
- `requiresWhitelist` added input on `createProduct`
- `requiresWhitelist` added output on `getProduct`
- `buyFor` added. Required for Uniswap Adaptor contract.
- Whitelist functions added.

AFAIK no whitelist functionality has been added to the E&E backend yet.

This PR will require you to get the latest `docker pull streamr/streamr-ganache:dev` 
Note, once updated, the local watcher should be pointing to the new marketplace contract address.

If transactions fail, try restarting the browser and reseting Metamask. 

I'm not sure how to update/deploy the new MP contract on staging - I will follow up on this before merging.

### CODE REVIEW

**Comment by fonty1** [comment]
FYI it's been decided that the ABI will revert, so I will amend this PR soon.

**Comment by fonty1** [comment]
@juhah publishing now works here with `NEW_MP_CONTRACT=on` & `NEW_MP_CONTRACT=undefined`. 

Travis config'd to use the new contract, except for production as the new MP contract does not exist on main net AFAIK.

**Comment by fonty1** [comment]
@juhah 
I took out `NEW_MP_CONTRACT` (you were right, not a good place for it) from `.env.required`. Also removed `COMMUNITY_PRODUCTS=on` which I think snuck in from the uniswap PR.  

Added: DAI Main net contract:
`0x6B175474E89094C44Da98b954EedeAC495271d0F`

Address proofs:
https://etherscan.io/address/0x6B175474E89094C44Da98b954EedeAC495271d0F

https://github.com/makerdao/developerguides/blob/master/mcd/upgrading-to-multi-collateral-dai/upgrading-to-multi-collateral-dai.md#as-a-centralized-exchange-or-custodial-wallet

Note in pull requests, `NEW_MP_CONTRACT` is on, but since it's the mainnet, the uniswap functionality won't be available.

**Comment by fonty1** [comment]
Bit easier to see it all together @juhah ,

**Local**:
**DATA** 0xbAA81A0179015bE47Ad439566374F2Bae098686F
**DAI** 0x642d2b84a32a9a92fec78ceaa9488388b3704898
**MP1** 0x0af64558670a3b761B57e465Cb80B62254b39619
**MP2** 0xF1371c0f40528406dc4f4cAf89924eA9Da49E866
**UNISWAP** 0xe4ea76e830a659282368ca2e7e4d18c4ae52d8b3
**NEWCONTRACT**=yes
**Source**: https://github.com/streamr-dev/streamr-ganache/blob/master/README.md

**Staging**:
**DATA** 0x67A9FDA28c4Df570D1d5350ACfA345194F4d4100
**DAI** 0x4CEed45382203e84d2eAB4A9F37D4F141e79A574
**MP1** 0x62D08A44D6DB7821C9556032A437BEB4bD050ABD
**MP2** 0x98af5996A5cbfAC95aE1E860cBD60d472f214Be0
**UNISWAP** 0x207c73d29F12e28c4e08C82A9d0f6Fb38CEa60C3
**NEWCONTRACT**=yes
**Source**: https://github.com/streamr-dev/streamr-infra/wiki/Private-Ethereum-Chain

**Prod**:
**DATA** 0x0Cf0Ee63788A0849fE5297F3407f701E122cC023
**DAI** 0x6B175474E89094C44Da98b954EedeAC495271d0F
**MP1** 0xA10151D088f6f2705a05d6c83719e99E079A61C1
**MP2** N/A
**UNISWAP** N/A
**NEWCONTRACT**=no
**Source**:
https://etherscan.io/address/0x6B175474E89094C44Da98b954EedeAC495271d0F
https://github.com/streamr-dev/streamr-platform/blob/development/.travis.yml

File: .travis.yml [hunk 1]
```diff
@@ -174,28 +177,30 @@ jobs:
...
+    - DAI_TOKEN_CONTRACT_ADDRESS=0x4CEed45382203e84d2eAB4A9F37D4F141e79A574
+    - UNISWAP_ADAPTOR_CONTRACT_ADDRESS=0x207c73d29F12e28c4e08C82A9d0f6Fb38CEa60C3
+    - WEB3_REQUIRED_NETWORK_ID=8995
+    - WEB3_PUBLIC_HTTP_PROVIDER=https://staging.streamr.com:8540
+    - WEB3_PUBLIC_WS_PROVIDER=wss://staging.streamr.com:8540/ws
     - SENTRY_DSN=https://e012761929d24baa96ca82772f60b395@sentry.io/1203154
     - LOGROCKET_SLUG=et0uhu/streamr-frontend-local
     - COMMUNITY_PRODUCTS=on
     - STREAMR_ENGINE_NODE_ADDRESSES=0xFCAd0B19bB29D4674531d6f115237E16AfCE377c
-    - COMMUNITY_PRODUCTS=on
```
Comment (fonty1) [inline .travis.yml]: no community products on PR Branches?

File: .travis.yml [hunk 2]
```diff
@@ -174,28 +177,30 @@ jobs:
     if: type IN (pull_request)
     env:
     - NODE_ENV=production
-    - BUNDLE_ANALYSIS=1
     - PORT=12345
-    - VERSION=$TRAVIS_PULL_REQUEST_BRANCH
-    - SENTRY_ENVIRONMENT=pr
+    - VERSION=$TRAVIS_BRANCH
+    - SENTRY_ENVIRONMENT=staging
```
Comment (fonty1) [inline .travis.yml]: If this is the PR branch builds, why change these?
    - VERSION=$TRAVIS_PULL_REQUEST_BRANCH
    - SENTRY_ENVIRONMENT=pr

File: .travis.yml [hunk 3]
```diff
@@ -170,26 +177,30 @@ jobs:
...
+    - BUNDLE_ANALYSIS=1
     - STREAMR_API_URL=https://staging.streamr.com/api/v1
     - STREAMR_WS_URL=wss://staging.streamr.com/api/v1/ws
     - PLATFORM_ORIGIN_URL=https://staging.streamr.com
     - STREAMR_URL=https://staging.streamr.com
+    - PLATFORM_PUBLIC_PATH=https://marketplace-staging.streamr.com
```
Comment (fonty1) [inline .travis.yml]: I think this is OK, let's double check it all still works after merge. I can't remember the exact magic that makes these kind of builds happy 🙈

**Comment by fonty1** [comment]
I'm trying to find my commit that pushed those env vars to the travis file already- I must have switched branches and left it behind! Sorry didn't mean for you to have to do them. 
All looks good now, a few very minor comments, merge this as you please and show us the 🟢light :)

File: .travis.yml (hunk 2 above)
Comment (juhah) [reply .travis.yml]: ah, copy paste error - I should have left these in 🤦‍♂

File: .travis.yml (hunk 1 above)
Comment (juhah) [reply .travis.yml]: It's there on line 202, this got thrown by the copy-paste

File: app/src/shared/web3/config.js [hunk 4]
```diff
@@ -33,8 +34,10 @@ const getConfig = (): Config => ({
         address: process.env.DAI_TOKEN_CONTRACT_ADDRESS || '',
     },
     marketplace: {
-        abi: marketplaceAbi,
-        address: process.env.MARKETPLACE_CONTRACT_ADDRESS || '',
+        abi: process.env.NEW_MP_CONTRACT ? marketplaceAbi : marketplaceAbiOld,
+        address: (process.env.NEW_MP_CONTRACT ?
```
Comment (juhah) [inline app/src/shared/web3/config.js]: And as a note, the vars need to be evaluated inside the method instead of the top level because `process.env` vars are overwritten in the test before calling the method but *after* loading this file.

File: app/test/unit/components/Notifications.test.jsx [hunk 5]
```diff
@@ -188,7 +188,7 @@ describe(Notifications, () => {
         })
 
         it('removes notification instances from the state if isModalOpen is false', () => {
-            const notifications = mount(<Notifications />)
+            const notifications = mount(<Notifications noAnimation />)
```
Comment (juhah) [inline app/test/unit/components/Notifications.test.jsx]: Without `noAnimation` flag, `NotificationSystem` doesn't call the `onRemove` method

File: .travis.yml (hunk 2 above)
Comment (fonty1) [reply .travis.yml]: Yep. We switched a while back to be 'more production settings like' but now with a proper private blockchain I think it's a good time to switch back to be like staging.

File: app/src/shared/web3/config.js (hunk 4 above)
Comment (fonty1) [reply app/src/shared/web3/config.js]: Ahh, interesting.

File: .travis.yml (hunk 2 above)
Comment (juhah) [reply .travis.yml]: Ok, please push those new lines if you are making changes to here or let me know if I should fix them 👍

Comment (fonty1) [reply .travis.yml]: Just doing a last sanity check, then I'll push those lines and merge.

Comment (juhah) [reply .travis.yml]: Perfect, thanks 🎉
